### PR TITLE
Move CORS functionalities into a separate class

### PR DESCRIPTION
**Pull request by sachithKay**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/synapse/rest/cors/SynapseCORSConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        String allowedOriginListStr =
+                SynapsePropertiesLoader.getPropertyValue(RESTConstants.CORS_CONFIGURATION_ACCESS_CTL_ALLOW_ORIGIN, "");
+        allowedOrigins = new HashSet<>(Arrays.asList(allowedOriginListStr.split(",")));
+
+        //Retrieve allowed headers
+        allowedHeaders =
+                SynapsePropertiesLoader.getPropertyValue(RESTConstants.CORS_CONFIGURATION_ACCESS_CTL_ALLOW_HEADERS, "");
+    }
+
+    public static SynapseCORSConfiguration getCORSConfig() {
```
Comment (milindaperera) [inline modules/core/src/main/java/org/apache/synapse/rest/cors/SynapseCORSConfiguration.java]: Let's rename this function as getInstance as common practice

File: modules/core/src/main/java/org/apache/synapse/rest/cors/CORSHelper.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.rest.RESTConstants;
+
+import java.util.Map;
+import java.util.Set;
+
+public class CORSHelper {
```
Comment (milindaperera) [inline modules/core/src/main/java/org/apache/synapse/rest/cors/CORSHelper.java]: Let's add class description

File: modules/core/src/main/java/org/apache/synapse/rest/cors/SynapseCORSConfiguration.java (hunk 1 above)
Comment (sachithKay) [reply modules/core/src/main/java/org/apache/synapse/rest/cors/SynapseCORSConfiguration.java]: fixed

File: modules/core/src/main/java/org/apache/synapse/rest/cors/CORSHelper.java (hunk 2 above)
Comment (sachithKay) [reply modules/core/src/main/java/org/apache/synapse/rest/cors/CORSHelper.java]: fixed

**Pull request merged**